packages/concerto-core/lib/introspect/stringvalidator.js

```diff
@@ -33,10 +33,18 @@ class StringValidator extends Validator{
      * @throws {IllegalModelException}
      */
     constructor(field, validator) {
-        super(field,validator);
+        super(field, validator);
         try {
-            // discard the leading / and closing /
-            this.regex = new RegExp(validator.substring(1,validator.length-1));
+            // Extract the RegExp components (pattern and flag)
+            const match = validator.match(/^\/([^/]*)\/(.*)$/);
```

Comment:
I think that this will fail for regexes that include escaped slashed in the pattern, e.g. `/foo\/bar/u`
![image](https://user-images.githubusercontent.com/7544022/128206210-d0c9abeb-1b36-4d4c-aec5-9058c3b66bef.png)

Reply:
yep! good catch.